packages/convai-widget-core/src/markdown/components/components.tsx

```diff
@@ -0,0 +1,720 @@
+  ({ children, className, node, ...props }: UlProps) => (
+    <ul
+      className={cn(
+        "ml-4 list-outside list-disc whitespace-normal text-sm",
+        className
+      )}
+      data-streamdown="unordered-list"
+      {...props}
+    >
+      {children}
+    </ul>
+  ),
+  (p, n) => sameClassAndNode(p, n)
+);
+MemoUl.displayName = "MarkdownUl";
+
+type HrProps = WithNode<JSX.IntrinsicElements["hr"]>;
+const MemoHr = memo<HrProps>(
+  ({ className, node, ...props }: HrProps) => (
+    <hr
+      className={cn("my-6 border-base-border", className)}
+      data-streamdown="horizontal-rule"
+      {...props}
+    />
+  ),
+  (p, n) => sameClassAndNode(p, n)
+);
+MemoHr.displayName = "MarkdownHr";
+
+type StrongProps = WithNode<JSX.IntrinsicElements["span"]>;
+const MemoStrong = memo<StrongProps>(
+  ({ children, className, node, ...props }: StrongProps) => (
+    <span
+      className={cn("font-medium", className)}
+      data-streamdown="strong"
+      {...props}
+    >
+      {children}
+    </span>
+  ),
+  (p, n) => sameClassAndNode(p, n)
+);
+MemoStrong.displayName = "MarkdownStrong";
+
+type AProps = WithNode<JSX.IntrinsicElements["a"]> & { href?: string };
+const MemoA = memo<AProps>(
+  ({ children, className, href, node, ...props }: AProps) => {
+    const isIncomplete = href === "streamdown:incomplete-link";
+
+    return (
+      <a
+        className={cn("wrap-anywhere font-medium underline", className)}
+        data-incomplete={isIncomplete}
+        data-streamdown="link"
+        href={href}
+        rel="noreferrer"
+        target="_blank"
+        {...props}
+      >
+        {children}
+      </a>
+    );
+  },
+  (p, n) => sameClassAndNode(p, n) && p.href === n.href
+);
+MemoA.displayName = "MarkdownA";
+
+type HeadingProps<TTag extends keyof JSX.IntrinsicElements> = WithNode<
+  JSX.IntrinsicElements[TTag]
+>;
+
+const MemoH1 = memo<HeadingProps<"h1">>(
+  ({ children, className, node, ...props }) => (
+    <h1
+      className={cn(
+        "mt-6 mb-2 text-base-primary font-semibold text-xl",
+        className
+      )}
+      data-streamdown="heading-1"
+      {...props}
+    >
+      {children}
+    </h1>
+  ),
+  (p, n) => sameClassAndNode(p, n)
+);
+MemoH1.displayName = "MarkdownH1";
+
+const MemoH2 = memo<HeadingProps<"h2">>(
+  ({ children, className, node, ...props }) => (
+    <h2
+      className={cn(
+        "mt-3 mb-2 text-base-primary font-semibold text-lg",
+        className
+      )}
+      data-streamdown="heading-2"
+      {...props}
+    >
+      {children}
+    </h2>
+  ),
+  (p, n) => sameClassAndNode(p, n)
+);
+MemoH2.displayName = "MarkdownH2";
+
+const MemoH3 = memo<HeadingProps<"h3">>(
+  ({ children, className, node, ...props }) => (
+    <h3
+      className={cn(
+        "mt-3 mb-1 text-base-primary font-semibold text-md",
+        className
+      )}
+      data-streamdown="heading-3"
+      {...props}
+    >
+      {children}
+    </h3>
+  ),
+  (p, n) => sameClassAndNode(p, n)
```

Comment:
I'm guessing this is copy-paste from streamdown but could be just:
```suggestion
  sameClassAndNode
```